.travis.yml

```diff
@@ -6,10 +8,68 @@ cache:
     directories:
         - "~/.platformio"
 
+# Configuration variables.
 env:
-    - PLATFORMIO_CI_SRC=src
+  global:
+    - ROS_DISTRO=indigo
+    - ROS_CI_DESKTOP="`lsb_release -cs`"  # e.g. [precise|trusty|...]
+    - CI_SRC_PATH=$(pwd)  
+  matrix:
+    - PLATFORMIO_CI_SRC=$CI_SRC_PATH/Arduino/victoria_teensy_ros_node
+    - PLATFORMIO_CI_SRC=$CI_SRC_PATH/Arduino/victoria_teensy_circuit_test_ros_node
+    - PLATFORMIO_CI_SRC=$CI_SRC_PATH/Arduino/victoria_trex_ros_test
+    - PLATFORMIO_CI_SRC=$CI_SRC_PATH/Arduino/teensy_tinygps_test
+
+before_install:
+    # Get roskeys
+    - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
+    - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+    - sudo apt-get -qq update
+
 install:
+    # Install minimum amount of ROS to get things working for rosserial.
+    - sudo apt-get install -y ros-$ROS_DISTRO-catkin
+    - sudo apt-get install -y python-rosdep
+    - sudo apt-get install -y ros-$ROS_DISTRO-rosbash
+    - sudo apt-get install -y ros-$ROS_DISTRO-roscpp
+    - sudo apt-get install -y ros-$ROS_DISTRO-roslib
+    - sudo apt-get install -y ros-$ROS_DISTRO-rospy
+    - sudo apt-get install -y ros-$ROS_DISTRO-common-msgs
+    - sudo apt-get install -y ros-$ROS_DISTRO-topic-tools
+    - sudo apt-get install -y ros-$ROS_DISTRO-tf
+    - pip install -U rospkg
+    - pip install -U catkin_pkg
+    - pip install -U empy
+    - source /opt/ros/$ROS_DISTRO/setup.bash
+    # Prepare rosdep to install dependencies.
+    - sudo rosdep init
+    - rosdep update
+    # Install platformio
     - pip install platformio
+    ### Get platformio libs.
+    # Install Timer.h library for easy ISRs.
+    # http://platformio.org/lib/show/75/Timer
+    # Pull in custom IMU libraries that use the i2c_t3.h instead of Time.h
+    - platformio lib -g install 75 https://github.com/victoriarobotics/lsm6-arduino.git https://github.com/victoriarobotics/lis3mdl-arduino.git
```

Comment:
Sorry, I missed the wrap around the first time around. I don't pretend to understand all the Travis stuff, but I see it is including the right libraries here at least.